cfme/fixtures/nuage.py

```diff
@@ -43,6 +44,27 @@ def create_basic_sandbox(nuage):
     box['l2_group'] = box['l2_domain'].create_child(
         nuage.vspk.NUPolicyGroup(name=random_name()))[0]
 
+    return create_namespace_from_enterprise(box)
+
+
+def create_namespace_from_enterprise(enterprise):
```

Comment:
Is there a reason to first create a hash `box` and then convert into object? Wouldn't it read better if we do object just right away in L10, like this:

```python
box = Namespace()

# Create empty enterprise aka 'sandbox'.
box.enterprise = dict(obj=nuage.create_enterprise())

# Fill the sandbox with some entities.
 # Method `create_child` returns a tuple (object, connection) and we only need object.
box.template = dict(
  obj=box.enterprise.create_child(nuage.vspk.NUDomainTemplate(name=random_name()))[0]
)
box.domain = dict(
  obj=enterprise.create_child(nuage.vspk.NUDomain(name=random_name(), template_id=box.template['obj'].id))[0],
  num_subnets=1,
  num_security_groups=1  
)
...

return box
```

Reply:
You are right, I created objects directly into `Namespace`. I also saved their names and ids, so we won't have to access 'obj' key everytime.

Reply:
Coming back to these older comments after having made some new ones. @miha-plesko I think your suggestion is on the right track, but its still not clear to me why the objects are being wrapped in dictionaries at all here. Can `box.enterprise` and `box.template` not be objects? Is this a Namespace limitation? If so, let's use something other than Namespace.

Reply:
@mshriver the idea here was to make it "flat" - the object itself comes directly from Nuage python library which we wanted to only use within this fixture. So we were thinking to create anonymous class (Namespace) and only copy attributes we are interested in later for tests, then drop the object itself.

Another reason why just passing on the object itself didn't seem a good idea was that it uses some fancy caching mechanism which in turn yield wrong values unless correctly handled in the test, which makes the whole thing less readable.

To make the story short: we want to use Nuage python library barely to CREATE some entities. We don't need to READ any of them, because we know exactly what we created, hence the plain simple Namespace. If we wanted to use the real object, then we would have to move it to wrapanapi, which is an overkill in this case, I think :)

Please let me know if that answers your question, we can also have a quick call to talk it out in person.